impl/hash_map.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, NamedTuple, Union
+
+from interface.hash_map import HashMapInterface
+
+
+class Pair(NamedTuple):
+    key: Any
+    value: Any
+
+
+class HashMap(HashMapInterface):
+    def __init__(self, size=8):
+        if size < 1:
+            raise ValueError("Size must be a positive number")
+        self._slots = size * [None]
+
+    def set(self, key, value):
+        for index, pair in self._probe(key):
+            if pair is None or pair.key == key:
+                self._slots[index] = Pair(key, value)
+                break
+        else:
+            self._resize_and_rehash()
+            self.set(key, value)
+
+    def get(self, key):
+        for _, pair in self._probe(key):
+            if pair is None:
+                raise KeyError(key)
+            if pair.key == key:
+                return pair.value
+        raise KeyError(key)
+
+    def _resize_and_rehash(self):
+        copy = HashMap(size=self.size * 2)
+        for key, value in self._slots:
+            copy.set(key, value)
+        self._slots = copy._slots
```

Comment:
If you decide to copy params from another instance, you should copy all the params: `self.__dict__ = copy.__dict__`  
Otherwise we can get unexpected behaviour once new property added